docs/developers/components/managing.md

```diff
@@ -0,0 +1,162 @@
+
+Stop execution for either of these processes by sending a SIGINT (generally CTRL/CMD+C) signal to the process.
+
+### Deploying to a local Harper instance
+
+Alternatively, to mimic interfacing with a hosted Harper instance, use operation commands instead.
+
+1. Start up Harper with `harperdb`
+2. _Deploy_ the component to the local instance by executing:
+    ```sh
+    harperdb deploy_component \
+      project=<name> \
+      package=<path-to-project> \
+      restart=true \
+      replicated=true
+    ```
+   - Make sure to omit the `target` option so that it _deploys_ to the Harper instance running locally
+   - The `package=<path>` option creates a symlink to the component simplifying restarts
+   - The `restart=true` option automatically restarts Harper threads after the component is deployed
+   - The `replicated=true` option is important if running on multiple threads, but optional otherwise
+3. In another process, use the `harperdb restart` command to restart the instance's threads at any time.
+   - With `package=<path>`, the component source is symlinked so changes will automatically be picked up between restarts!
+   - If `package` was omitted, run the `deploy_component` command again with any new changes.
+4. To remove the component use `harperdb drop_component project=<name>`
+
+Similar to the previous section, if the main thread needs to be restarted, start and stop the Harper instance manually (with the component deployed). Upon Harper startup, the component will automatically be loaded and executed across all threads.
+
+> Not all [component operations](../operations-api/components.md) are available via CLI. When in doubt, switch to using the Operations API via network requests to the local Harper instance.
+
+For example, to properly _deploy_ a `test-component` locally, the command would look like:
+
+```sh
+harperdb deploy_component \
+  project=test-component \
+  package=/Users/dev/test-component \
+  restart=true \
+  replicated=true
+```
+
+> [!TIP]
+> If the current directory is the component directory, use a shortcut such as `package=$(pwd)` to avoid typing out the complete path.
+
+## Remote Management
+
+Managing components on a remote Harper instance is best accomplished through [component operations](../operations-api/components.md), similar to using the `deploy_component` command locally. Before continuing, always backup critical Harper instances. Managing, deploying, and executing components can directly impact a live system.
+
+Remote Harper instances work very similarly to local Harper instances. The primary component management operations still include `deploy_component`, `drop_component`, and `restart`.
+
+The key to remote management is using the `CLI_TARGET_USERNAME` and `CLI_TARGET_PASSWORD` environment variables, and the `target=<remote>` CLI option.
```

Comment:
FWIW, you can  also use `username=<user>` and `password=<pass>` as command line arguments. I just prefer env variables, so that's what I have always suggested to people.

Reply:
I will include this as an option then!